cmd/otel-allocator/prehook/relabel_test.go

```diff
@@ -30,77 +30,134 @@ import (
+					Action:       "keep",
+					Replacement:  "$1",
+				},
 			},
 			isDrop: false,
 		},
 		{
-			cfg: relabel.Config{
-				SourceLabels: model.LabelNames{"i"},
-				Regex:        relabel.MustNewRegexp("bad.*match"),
-				Action:       "drop",
+			cfg: []*relabel.Config{
+				{
+					SourceLabels: model.LabelNames{"i"},
+					Regex:        relabel.MustNewRegexp("bad.*match"),
+					Action:       "drop",
+					Separator:    ";",
+					Replacement:  "$1",
+				},
 			},
 			isDrop: false,
 		},
 		{
-			cfg: relabel.Config{
-				SourceLabels: model.LabelNames{"label_not_present"},
-				Regex:        relabel.MustNewRegexp("(.*)"),
-				Action:       "keep",
+			cfg: []*relabel.Config{
+				{
+					SourceLabels: model.LabelNames{"label_not_present"},
+					Regex:        relabel.MustNewRegexp("(.*)"),
+					Separator:    ";",
+					Action:       "keep",
+					Replacement:  "$1",
+				},
 			},
 			isDrop: false,
 		},
 		{
-			cfg: relabel.Config{
-				SourceLabels: model.LabelNames{"i"},
-				Regex:        relabel.MustNewRegexp("(.*)"),
-				Action:       "drop",
+			cfg: []*relabel.Config{
+				{
+					SourceLabels: model.LabelNames{"i"},
+					Regex:        relabel.MustNewRegexp("(.*)"),
+					Separator:    ";",
+					Action:       "drop",
+					Replacement:  "$1",
+				},
 			},
 			isDrop: true,
 		},
 		{
-			cfg: relabel.Config{
-				SourceLabels: model.LabelNames{"collector"},
-				Regex:        relabel.MustNewRegexp("(collector.*)"),
-				Action:       "drop",
+			cfg: []*relabel.Config{
+				{
+					SourceLabels: model.LabelNames{"collector"},
+					Regex:        relabel.MustNewRegexp("(collector.*)"),
+					Separator:    ";",
+					Action:       "drop",
+					Replacement:  "$1",
+				},
 			},
 			isDrop: true,
 		},
 		{
-			cfg: relabel.Config{
-				SourceLabels: model.LabelNames{"i"},
-				Regex:        relabel.MustNewRegexp("bad.*match"),
-				Action:       "keep",
+			cfg: []*relabel.Config{
+				{
+					SourceLabels: model.LabelNames{"i"},
+					Regex:        relabel.MustNewRegexp("bad.*match"),
+					Separator:    ";",
+					Action:       "keep",
+					Replacement:  "$1",
+				},
 			},
 			isDrop: true,
 		},
 		{
-			cfg: relabel.Config{
-				SourceLabels: model.LabelNames{"collector"},
-				Regex:        relabel.MustNewRegexp("collectors-n"),
-				Action:       "keep",
+			cfg: []*relabel.Config{
+				{
+					SourceLabels: model.LabelNames{"collector"},
+					Regex:        relabel.MustNewRegexp("collectors-n"),
+					Separator:    ";",
+					Action:       "keep",
+					Replacement:  "$1",
+				},
 			},
 			isDrop: true,
 		},
 	}
 
+	HashmodConfig = relabelConfigObj{
+		cfg: []*relabel.Config{
+			{
+				SourceLabels: model.LabelNames{"i"},
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Separator:    ";",
+				Modulus:      1,
+				TargetLabel:  "tmp",
+				Action:       "hashmod",
+				Replacement:  "$1",
+			},
+
+			{
+				SourceLabels: model.LabelNames{"tmp"},
+				Regex:        relabel.MustNewRegexp("$(SHARD)"),
```

Comment:
Is `$(SHARD)` only expected in the Regex or it can be in other fields as well?

Reply:
Thanks for the review! AFAIK, `$(SHARD)` is used by Prometheus in the generated config when using the `hashmod` action.  In this case I've only seen it as a Regex field.  Unsure if there would ever be a case where it would show up in a label. Just in case I added it to the `SourceLabels` as another confirmation that substitution is working as expected.